gum/backend-linux/gumandroid.c

```diff
@@ -666,19 +670,32 @@ gum_try_parse_linker_proc_maps_line (const gchar * line,
 
   if (sizeof (gpointer) == 4)
   {
-    new_path = "/apex/com.android.runtime/bin/linker";
-    old_path = "/system/bin/linker";
+    linker_expected_path = "/system/bin/linker";
   }
   else
   {
-    new_path = "/apex/com.android.runtime/bin/linker64";
-    old_path = "/system/bin/linker64";
+    linker_expected_path = "/system/bin/linker64";
   }
 
-  if (strcmp (path, new_path) == 0)
-    linker_path = new_path;
-  else if (strcmp (path, old_path) == 0)
-    linker_path = old_path;
+  // If there are no linker files, fail
+  if(lstat(linker_expected_path, &linker_stat) == -1) {
```

Comment:
Style should look like this:

```suggestion
  if (lstat (linker_expected_path, &linker_stat) == -1)
  {
```

Though in case of a single statement – unless the statement needs to be split across multiple lines – curly braces should be omitted.

(Sorry for not documenting it anywhere yet.)

Reply:
However, it would be good to use GLib's [file utilities](https://developer.gnome.org/glib/stable/glib-File-Utilities.html) here:

```C
  if (g_file_test (linker_expected_path, G_FILE_TEST_IS_SYMLINK))
  {
    …
  }
```

And [g_file_read_link()](https://developer.gnome.org/glib/stable/glib-File-Utilities.html#g-file-read-link) to read the link. (Return value needs to be `g_free()`d.)

Reply:
Makes sense, will switch to using glib's file utils and fix the styling.